### PR TITLE
feat/english & chinese translation redemption stats

### PR DESCRIPTION
**Pull request by shermainelim**
This PR adds... 

Related to: [Chinese translation for app Redemption Statistics](https://www.notion.so/701364d4fbbf4ec1801b766c589973b9?v=24aa9675206d4da9a171ca4073d81b5b&p=1cbd66b827b74892a1d072109b438a47)

-call i18ntWithValidator directly for util.ts via Appeal as outside react component

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [x] I've requested reviews from 1-2 reviewers
- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [x] I've added/updated unit/integration tests
- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
- [x] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

Old : https://github.com/rationally-app/mobile-application/pull/232

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,14 +0,0 @@
...
-<!-- This checklist is just a guideline.-->
-<!-- If any of the following does not apply to your PR, please leave them unchecked and explain in the PR description -->
-
-- [ ] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
-- [ ] I've requested reviews from 1-2 reviewers
-- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
-- [ ] I've added/updated unit/integration tests
-- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
-- [ ] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
-- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)
```
Comment (sebastianquek) [inline .github/pull_request_template.md:14]: This shouldn't be deleted ya

Comment (shermainelim) [reply .github/pull_request_template.md:14]: done

Comment (sebastianquek) [reply .github/pull_request_template.md:14]: hmm? it's still committed, do check under files changed!

File: src/common/i18n/translations/zh.ts [hunk 2]
```diff
@@ -1,6 +1,15 @@
 import { Translations } from "./type";
 
 export const zh: Translations = {
+  redemptionStats: {
+    youDistributed: "您已分发",
+    lastDistributedAt: "最后在 %{lastTransactionTime} 分发",
+    back: " 返回",
+    statistics: "统计",
+    goToStatistics: "前往统计",
+    noItemsScanned: "没有扫描项目",
```
Comment (ilhaoni) [inline src/common/i18n/translations/zh.ts:10]: Should it be `没有分发记录`?

File: src/common/i18n/translations/zh.ts [hunk 3]
```diff
@@ -1,6 +1,15 @@
 import { Translations } from "./type";
 
 export const zh: Translations = {
+  redemptionStats: {
+    youDistributed: "您已分发",
+    lastDistributedAt: "最后在 %{lastTransactionTime} 分发",
+    back: " 返回",
+    statistics: "统计",
+    goToStatistics: "前往统计",
```
Comment (ilhaoni) [inline src/common/i18n/translations/zh.ts:9]: Would `查看统计` be better?

**Review by ilhaoni — COMMENTED** [review]
Have some comments about the translations!

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx [hunk 4]
```diff
@@ -4,10 +4,12 @@ import { CampaignConfigContext } from "../../context/campaignConfig";
 import { getDailyStatistics } from "../../services/statistics";
 import { countTotalTransactionsAndByCategory } from "./utils";
 import { sortTransactionsByOrder } from "../../components/CustomerQuota/utils";
+import { useTranslate } from "../useTranslate/useTranslate";
 
 export type StatisticsHook = {
   totalCount: number | null;
   lastTransactionTime: Date | null;
+
```
Comment (sebastianquek) [inline src/hooks/useDailyStatistics/useDailyStatistics.tsx:12]:
```suggestion
```

File: src/components/DailyStatistics/TransactionHistoryCard.tsx [hunk 5]
```diff
@@ -13,7 +14,6 @@ interface TransactionHistoryCardComponent {
   }[];
   loading: boolean;
 }
-
```
Comment (sebastianquek) [inline src/components/DailyStatistics/TransactionHistoryCard.tsx:16]: Nit, please revert this. So only changes related to this feature is reflected in the version history.

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 4 above)
Comment (sebastianquek) [inline src/hooks/useDailyStatistics/useDailyStatistics.tsx:12]: Nit: please remove this new line as well

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx [hunk 6]
```diff
@@ -35,10 +37,9 @@ export const useDailyStatistics = (
   const [transactionHistory, setTransactionHistory] = useState<
     StatisticsHook["transactionHistory"]
   >([]);
-
+  const { c13ntForUnit } = useTranslate();
   const { policies } = useContext(CampaignConfigContext);
   const prevTimestamp = usePrevious(currentTimestamp);
-
```
Comment (sebastianquek) [inline src/hooks/useDailyStatistics/useDailyStatistics.tsx:41]: nit: this too.

File: src/hooks/useDailyStatistics/utils.ts [hunk 7]
```diff
@@ -1,7 +1,9 @@
 import { chain, sumBy } from "lodash";
-import { DailyStatistics, CampaignPolicy } from "../../types";
 import { formatQuantityText } from "../../components/CustomerQuota/utils";
+import { DailyStatistics, CampaignPolicy } from "../../types";
+
```
Comment (sebastianquek) [inline src/hooks/useDailyStatistics/utils.ts:4]: Not sure why there's this change, when the import is already there

**Review by sebastianquek — CHANGES_REQUESTED** [review]
Hmm noticed that you renamed utils.test
` src/hooks/useDailyStatistics/utils.test.ts → src/hooks/useDailyStatistics/utils.test.tsx`
Given it doesn't render any jsx, why should it be .tsx?

File: src/common/i18n/translations/zh.ts (hunk 3 above)
Comment (shermainelim) [reply src/common/i18n/translations/zh.ts:9]: done

File: src/components/DailyStatistics/TransactionHistoryCard.tsx (hunk 5 above)
Comment (shermainelim) [reply src/components/DailyStatistics/TransactionHistoryCard.tsx:16]: done

File: .github/pull_request_template.md (hunk 1 above)
Comment (shermainelim) [reply .github/pull_request_template.md]: done

File: src/common/i18n/translations/zh.ts (hunk 2 above)
Comment (shermainelim) [reply src/common/i18n/translations/zh.ts]: done

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 4 above)
Comment (shermainelim) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx]: done

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 6 above)
Comment (shermainelim) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:41]: done

File: src/hooks/useDailyStatistics/utils.ts (hunk 7 above)
Comment (shermainelim) [reply src/hooks/useDailyStatistics/utils.ts:4]: rearranged, removed line 4 spacing

**Comment by shermainelim** [comment]
> Hmm noticed that you renamed utils.test
> ` src/hooks/useDailyStatistics/utils.test.ts → src/hooks/useDailyStatistics/utils.test.tsx`
> Given it doesn't render any jsx, why should it be .tsx?

done

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -5,10 +5,10 @@ This PR adds... <!-- A brief description of what your PR does -->
...
-- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
-- [ ] I've added/updated unit/integration tests
-- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
-- [ ] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
+- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
+- [x] I've requested reviews from 1-2 reviewers
+- [x] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
+- [x] I've added/updated unit/integration tests
+- [x] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
+- [x] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
```
Comment (sebastianquek) [inline .github/pull_request_template.md:13]: What's with these changes :O

Comment (shermainelim) [reply .github/pull_request_template.md:13]: i thought u meant get back the template?

File: src/common/i18n/translations/zh.ts (hunk 3 above)
Comment (ilhaoni) [reply src/common/i18n/translations/zh.ts:9]: Ehh, we should be changing `goToStatistics`.

I think you made the change on the wrong field haha

Comment (shermainelim) [reply src/common/i18n/translations/zh.ts]: oops done

File: .github/pull_request_template.md (hunk 8 above)
Comment (shermainelim) [reply .github/pull_request_template.md]: done

File: src/common/i18n/translations/en.ts [hunk 9]
```diff
@@ -1,6 +1,15 @@
 import { Translations } from "./type";
 
 export const en: Translations = {
+  redemptionStats: {
+    youDistributed: "You distributed",
+    lastDistributedAt: "Last distributed at %{lastTransactionTime}",
+    back: " Back",
```
Comment (sebastianquek) [inline src/common/i18n/translations/en.ts:7]: Noticed there's a space inside this, could you share more why this is needed?

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx [hunk 10]
```diff
@@ -83,6 +87,7 @@ export const useDailyStatistics = (
     sessionToken,
     currentTimestamp,
     prevTimestamp,
+    c13ntForUnit,
```
Comment (sebastianquek) [inline src/hooks/useDailyStatistics/useDailyStatistics.tsx:90]: I recall mentioning this previously, why does this need to be exported?

**Review by ilhaoni — APPROVED** [review]
Thanks for fixing the translations :D

**Comment by sebastianquek** [comment]
Please rebase this on master too so that the latest checks can be done!

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 10 above)
Comment (shermainelim) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:90]: need it for the screen to translate the uni

**Comment by shermainelim** [comment]
updated latest from master.

File: src/hooks/useDailyStatistics/useDailyStatistics.test.tsx [hunk 11]
```diff
@@ -7,6 +7,7 @@ import { defaultIdentifier } from "../../test/helpers/defaults";
 import { CampaignPolicy } from "../../types";
 import { Sentry } from "../../utils/errorTracking";
 import { wait } from "@testing-library/react-native";
+import "../../common/i18n/i18nMock";
```
Comment (ilhaoni) [inline src/hooks/useDailyStatistics/useDailyStatistics.test.tsx:10]: hmm, are we using this import here? if we're not using it, we can remove it haha (:

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 10 above)
Comment (ilhaoni) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:90]: hmm, I looked through the current repo, we dont import this useTranslate in other hooks. We currently import it in components that needs it haha. 

Do refer to other components, like NoQuotaCard, Stepper, ItemMaxUnitLabel and CheckoutSuccessCard to see how it is being implemented so far haha.

Comment (ilhaoni) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:90]: Do you mind sharing which screens needs this also haha, thanks!

File: src/hooks/useDailyStatistics/utils.ts [hunk 12]
```diff
@@ -41,14 +45,17 @@ export const countTotalTransactionsAndByCategory = (
...
+              label: ` ${i18ntWithValidator(
+                "checkoutSuccessScreen",
+                "quantity"
+              )}`,
             }
           ),
           descriptionAlert:
             findItemByCategory?.categoryType === "APPEAL"
-              ? "via appeal"
+              ? i18ntWithValidator("redemptionStats", "viaAppeal")
```
Comment (ilhaoni) [inline src/hooks/useDailyStatistics/utils.ts:58]: we're importing `i18ntWithValidator` directly from the useTranslate hook, but we're passing in just `c13ntForUnit` through the params here.

is there a reason why it is inconsistent here?

**Review by ilhaoni — CHANGES_REQUESTED** [review]
Have some questions haha. Some implementation looks different from the rest of the repo at the moment, hence would like to clarify first haha

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 10 above)
Comment (ilhaoni) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:90]: @shermainelim have reopened this, as I have questions on this too haha

File: src/hooks/useDailyStatistics/useDailyStatistics.test.tsx (hunk 11 above)
Comment (shermainelim) [reply src/hooks/useDailyStatistics/useDailyStatistics.test.tsx:10]: we need that because of mocking. if not the test will fail.

Comment (ilhaoni) [reply src/hooks/useDailyStatistics/useDailyStatistics.test.tsx:10]: I don't see anything being used here. What is being used from this import?

Comment (shermainelim) [reply src/hooks/useDailyStatistics/useDailyStatistics.test.tsx:10]: the purpose of importing that is to set the default translation locale, if not error will be thrown when translating texts.

File: src/hooks/useDailyStatistics/utils.ts (hunk 12 above)
Comment (shermainelim) [reply src/hooks/useDailyStatistics/utils.ts:58]: i18ntWithValidator is called directly from useTranslate.tsx as it is not a hook. it is only exposed as a method while c13ntForUnit relies on the useTranslate hook that cannot be extracted due to dependency on useContext() from CampaignConfigContext

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 10 above)
Comment (ilhaoni) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:90]: hmm, i dont think we should import a hook inside another hook. all hooks should only be imported at the component level and passed down if needed haha.

if you noticed in my previous comment, all current implementations follow this format. You can refer to NoQuotaCard, Stepper, ItemMaxUnitLabel and CheckoutSuccessCard to take a look.

File: src/hooks/useDailyStatistics/useDailyStatistics.test.tsx (hunk 11 above)
Comment (ilhaoni) [reply src/hooks/useDailyStatistics/useDailyStatistics.test.tsx:10]: i see, okok

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 10 above)
Comment (ilhaoni) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:90]: I think you can call the `useTranslate` hook at the DailyStatistics components, like how it's done now in the TitleStatistic and TransactionHistoryCard

File: src/hooks/useDailyStatistics/utils.ts (hunk 12 above)
Comment (ilhaoni) [reply src/hooks/useDailyStatistics/utils.ts:58]: hmm, i think as mentioned above, do try to follow the other implementations thus far, as we shouldn't be importing a hook inside another hook (:

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 10 above)
Comment (shermainelim) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:84]: -Refactored the typing in useDailyStatistics Utils by adding a new Transaction type that can be used by other files so there is no need to always retype the same typing that was used in the old approach

- Formatting quantity and unit with c13forUnit in TransactionHistoryCard with formatText method. 

- Removed Jest as no longer need to mock the translation as it will be done in the TransactionHistoryCard instead of utils.

File: src/hooks/useDailyStatistics/utils.ts (hunk 12 above)
Comment (shermainelim) [reply src/hooks/useDailyStatistics/utils.ts]: hi @ilhaoni 
I've moved all the translation related code from utils to TransactionHistoryCard so that the TransactionHistoryCard will be responsible for performing the translation instead of utils.

File: src/hooks/useDailyStatistics/utils.ts [hunk 13]
```diff
@@ -1,16 +1,25 @@
...
+  | undefined;
+
+export type Transaction = {
+  name: string;
+  category: string;
+  quantity: number;
+  unitType: UnitType;
+  descriptionAlert?: "viaAppeal" | undefined;
+  order: number;
+};
```
Comment (ilhaoni) [inline src/hooks/useDailyStatistics/utils.ts:19]: Thanks for the effort to refactor, but this is actually not a transaction. It's a combination of policies and the past transaction history.

Naming this transaction would conflict with the rest of the Transaction type in the repo as well.

If you would like to try to refactor, maybe we can do it in another PR? For now, I think it would be good to remove it. (:

File: src/hooks/useDailyStatistics/utils.ts [hunk 14]
```diff
@@ -39,16 +48,12 @@ export const countTotalTransactionsAndByCategory = (
...
-          quantityText: formatQuantityText(
-            totalQuantityInCategory,
-            findItemByCategory?.quantity.unit || {
-              type: "POSTFIX",
-              label: " qty",
-            }
-          ),
+          quantity: totalQuantityInCategory,
+          unitType: findItemByCategory?.quantity.unit,
+          // categoryType: findItemByCategory?.categoryType
```
Comment (ilhaoni) [inline src/hooks/useDailyStatistics/utils.ts:53]: is there a reason why this is commented out?

File: src/hooks/useDailyStatistics/utils.ts [hunk 15]
```diff
@@ -39,16 +48,12 @@ export const countTotalTransactionsAndByCategory = (
...
           category: key,
-          quantityText: formatQuantityText(
-            totalQuantityInCategory,
-            findItemByCategory?.quantity.unit || {
-              type: "POSTFIX",
-              label: " qty",
-            }
-          ),
+          quantity: totalQuantityInCategory,
+          unitType: findItemByCategory?.quantity.unit,
```
Comment (ilhaoni) [inline src/hooks/useDailyStatistics/utils.ts:52]: I think the default fallback value should be here, rather than leaving it as `unitType: undefined`. That doesnt really make sense haha

File: src/hooks/useDailyStatistics/utils.ts [hunk 16]
```diff
@@ -1,16 +1,25 @@
...
 import { DailyStatistics, CampaignPolicy } from "../../types";
-import { formatQuantityText } from "../../components/CustomerQuota/utils";
 import { Sentry } from "../../utils/errorTracking";
 
+export type UnitType =
+  | {
+      type: "POSTFIX" | "PREFIX";
+      label: string;
+    }
+  | undefined;
```
Comment (ilhaoni) [inline src/hooks/useDailyStatistics/utils.ts:10]: the default unit was `{ type: "POSTFIX", label: " qty"}`, why are we changing the default value to undefined?

File: src/hooks/useDailyStatistics/useDailyStatistics.tsx (hunk 10 above)
Comment (ilhaoni) [reply src/hooks/useDailyStatistics/useDailyStatistics.tsx:84]: Thanks haha. I left some comments on the refactoring and changes. Actually for this comment, since we're moving it out of the useDailyStatistics hook, can we remove this line?

**Review by ilhaoni — COMMENTED** [review]
Left some comments (:

**Comment by ilhaoni** [comment]
Closing this as there's a newer PR that includes this
